source/grid/manifold.cc

```diff
@@ -62,6 +62,128 @@ get_new_point (const Quadrature<spacedim> &quad) const
 }
 
 
+
+template <>
+Tensor<1,2>
+Manifold<2, 2>::
+normal_vector (const typename Triangulation<2, 2>::face_iterator &face,
+               const Point<2> &p) const
+{
+  const int spacedim=2, dim=2;
+
+  Tensor<1,spacedim> tangent = ((p-face->vertex(0)).norm_square() > (p-face->vertex(1)).norm_square() ?
+                                get_tangent_vector(p, face->vertex(0)) :
+                                -get_tangent_vector(p, face->vertex(1)));
+  return cross_product_2d(tangent);
+}
+
+template<>
+Tensor<1,3>
+Manifold<3, 3>::
+normal_vector (const typename Triangulation<3, 3>::face_iterator &face,
+               const Point<3> &p) const
+{
+  const int spacedim=3, dim=3;
+  Tensor<1,spacedim> t1,t2;
+
+  // Take the difference between p and all four vertices
+  Tensor<1,spacedim> dp[4];
+  double dpns[4];
+  int min_index=-1;
+  double min_distance = 0;
+  for (unsigned int i=0; i<4; ++i)
+    {
+      dp[i] = p-face->vertex(i);
```

Comment:
I don't think you are using the `dp` array outside this loop. Just declare a scalar variable `dp` here (within the loop) that stores the distance vector and get rid of the array.